components/src/preact/components/tooltip.tsx

```diff
@@ -0,0 +1,33 @@
+import { flip, offset, shift } from '@floating-ui/dom';
+import { type FunctionComponent } from 'preact';
+import { useRef, useState } from 'preact/hooks';
+import { type JSXInternal } from 'preact/src/jsx';
+
+import { dropdownClass } from './dropdown';
+import { useFloatingUi } from '../shared/floating-ui/hooks';
+
+export type TooltipProps = {
+    content: string | (() => JSXInternal.Element);
+};
+
+const Tooltip: FunctionComponent<TooltipProps> = ({ children, content }) => {
+    const [showHelp, setShowHelp] = useState(false);
+    const referenceRef = useRef<HTMLDivElement>(null);
+    const floatingRef = useRef<HTMLDivElement>(null);
+
+    useFloatingUi(referenceRef, floatingRef, [offset(5), shift(), flip()]);
+    const tooltipContent = typeof content === 'string' ? <div>{content}</div> : content();
+
+    return (
+        <div className='relative'>
+            <div ref={referenceRef} onMouseEnter={() => setShowHelp(true)} onMouseLeave={() => setShowHelp(false)}>
+                {children}
+            </div>
+            <div ref={floatingRef} className={`${dropdownClass} ${showHelp ? '' : 'hidden'}`}>
+                {tooltipContent}
+            </div>
```

Comment:
I think I made it work with CSS, which has the advantage, that we can rely on browser-native hover features, such as toggling the hover state in the dev tools:
```suggestion
            <div className='peer' ref={referenceRef}>
                {children}
            </div>
            <div ref={floatingRef} className={`${dropdownClass} invisible peer-hover:visible`}>
                {tooltipContent}
            </div>
```

Then we also don't need the `showHelp` state.

Reply:
See https://tailwindcss.com/docs/hover-focus-and-other-states#styling-based-on-sibling-state.